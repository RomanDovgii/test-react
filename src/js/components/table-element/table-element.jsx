import React from "react";
import {tableElementType} from "../../types/types";

const TableElement = ({person}) => {
  const {id, firstName, lastName, email, phone} = person;

  return (
    <tr className="table__row">
      <td className="table__data">{id}</td>
      <td className="table__data">{firstName}</td>
      <td className="table__data">{lastName}</td>
      <td className="table__data">{email}</td>
      <td className="table__data">{phone}</td>
    </tr>
  );
};

TableElement.propTypes = tableElementType;

export default TableElement;
