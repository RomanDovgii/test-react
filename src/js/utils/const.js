export const MAXIMUM_USERS_PER_PAGE = 50;

export const AppRoute = {
  SELECT: `/select`,
  TABLE: `/table`
};
