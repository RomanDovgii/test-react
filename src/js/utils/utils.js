export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const sortToTop = (a, b) => {
  switch (true) {
    case a > b:
      return -1;
    case a < b:
      return 1;
    default:
      return 0;
  }
};

export const sortToBot = (a, b) => {
  switch (true) {
    case a < b:
      return -1;
    case a > b:
      return 1;
    default:
      return 0;
  }
};
