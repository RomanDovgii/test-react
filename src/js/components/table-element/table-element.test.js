import React from "react";
import renderer from "react-test-renderer";
import {TableElement} from "./table-element";
import {noop} from "../../utils/utils";

const mockUser = {
  id: 339,
  firstName: `Susen`,
  lastName: `Graham`,
  email: `LNicholson@mi.org`,
  phone: `(220)920-9427`,
  address: {
    streetAddress: `7422 Vitae Rd`,
    city: `Bergen County`,
    state: `NE`,
    zip: `24099`
  },
  description: `aenean augue neque mattis sollicitudin turpis sed aenean elit fringilla nec massa libero nullam lectus aenean et vel sed pulvinar et augue facilisis vitae libero dolor massa adipiscing dolor vitae orci tortor`
};

it(
    `Should render TableElement`,
    () => {
      const tree = renderer
      .create(
          <TableElement
            user={mockUser}
            onRowClick={noop}
          />
      )
      .toJSON();

      expect(tree).toMatchSnapshot();
    }
);
