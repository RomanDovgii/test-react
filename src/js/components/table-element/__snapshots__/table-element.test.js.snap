// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render TableElement 1`] = `
<tr
  className="table__row"
  onClick={[Function]}
>
  <td
    className="table__data"
  >
    339
  </td>
  <td
    className="table__data"
  >
    Susen
  </td>
  <td
    className="table__data"
  >
    Graham
  </td>
  <td
    className="table__data"
  >
    LNicholson@mi.org
  </td>
  <td
    className="table__data"
  >
    (220)920-9427
  </td>
</tr>
`;
